daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala

```diff
@@ -1434,15 +1453,30 @@ private[lf] object DecodeV1 {
         GEQ_NUMERIC,
         BGreaterEqNumeric,
         minVersion = numeric,
+        maxVersion = Some(genComparison),
       ),
-      BuiltinFunctionInfo(GEQ_TEXT, BGreaterEq, implicitParameters = List(TText)),
-      BuiltinFunctionInfo(GEQ_TIMESTAMP, BGreaterEq, implicitParameters = List(TTimestamp)),
+      BuiltinFunctionInfo(
+        GEQ_TEXT,
+        BGreaterEq,
+        maxVersion = Some(genComparison),
+        implicitParameters = List(TText)),
+      BuiltinFunctionInfo(
+        GEQ_TIMESTAMP,
+        BGreaterEq,
+        maxVersion = Some(genComparison),
+        implicitParameters = List(TTimestamp)),
       BuiltinFunctionInfo(
         GEQ_PARTY,
         BGreaterEq,
+        minVersion = partyOrdering,
+        maxVersion = Some(genComparison),
         implicitParameters = List(TParty),
-        minVersion = partyOrdering),
-      BuiltinFunctionInfo(LESS_INT64, BLess, implicitParameters = List(TInt64)),
+      ),
+      BuiltinFunctionInfo(
+        LESS_INT64,
+        BLess,
+//        maxVersion = Some(genComparison),
```

Comment:
Is there a reason for this one to be commented out?

Reply:
I was debugging the issue you fix in #5025.
I uncommented.